modules/gapi/test/serialization/serialization.cpp

```diff
@@ -0,0 +1,1292 @@
+    }
+}
+
+void cleanGSerializedOps(GSerialized &s)
+{
+
+//    std::cout << "cleanGSerializedOps" <<  std::endl;
+    for (uint i = 0; i < s.m_ops.size();  i++)
+    {
+        //Kernel k
+        std::fill(s.m_ops[i].k.name.begin(), s.m_ops[i].k.name.end(), 0);
+        s.m_ops[i].k.name.clear();
+        std::fill(s.m_ops[i].k.tag.begin(), s.m_ops[i].k.tag.end(), 0);
+        s.m_ops[i].k.tag.clear();
+
+        //std::vector<int>   kind;
+        std::fill(s.m_ops[i].kind.begin(), s.m_ops[i].kind.end(), 0);
+        s.m_ops[i].kind.clear();
+
+        //std::vector<int>   opaque_kind;
+        std::fill(s.m_ops[i].opaque_kind.begin(), s.m_ops[i].opaque_kind.end(), 0);
+        s.m_ops[i].opaque_kind.clear();
+
+        RcDesc zero_desc; zero_desc.id = 0; zero_desc.shape = GShape::GOPAQUE;
+        //std::vector<RcDesc> outs;
+        std::fill(s.m_ops[i].outs.begin(), s.m_ops[i].outs.end(), zero_desc);
+        s.m_ops[i].outs.clear();
+
+        //std::vector<RcDesc> ins;
+        std::fill(s.m_ops[i].ins.begin(), s.m_ops[i].ins.end(), zero_desc);
+        s.m_ops[i].ins.clear();
+
+        //opaque args
+        //std::vector<int> opaque_ints;
+        std::fill(s.m_ops[i].opaque_ints.begin(), s.m_ops[i].opaque_ints.end(), 0);
+        s.m_ops[i].opaque_ints.clear();
+
+        //std::vector<double> opaque_doubles;
+        std::fill(s.m_ops[i].opaque_doubles.begin(), s.m_ops[i].opaque_doubles.end(), 0.0);
+        s.m_ops[i].opaque_doubles.clear();
+
+        //std::vector<cv::Size> opaque_cvsizes;
+        std::fill(s.m_ops[i].opaque_cvsizes.begin(), s.m_ops[i].opaque_cvsizes.end(), cv::Size(0,0));
+        s.m_ops[i].opaque_cvsizes.clear();
+
+        //std::vector<bool> opaque_bools;
+        std::fill(s.m_ops[i].opaque_bools.begin(), s.m_ops[i].opaque_bools.end(), false);
+        s.m_ops[i].opaque_bools.clear();
+
+        //std::vector<cv::Scalar> opaque_cvscalars;
+        std::fill(s.m_ops[i].opaque_cvscalars.begin(), s.m_ops[i].opaque_cvscalars.end(), cv::Scalar(0,0,0,0));
+        s.m_ops[i].opaque_cvscalars.clear();
+
+        //std::vector<cv::Point> opaque_cvpoints;
+        std::fill(s.m_ops[i].opaque_cvpoints.begin(), s.m_ops[i].opaque_cvpoints.end(), cv::Point(0,0));
+        s.m_ops[i].opaque_cvpoints.clear();
+
+        //std::vector<cv::Mat> opaque_cvmats;
+        s.m_ops[i].opaque_cvmats.clear();
+
+        //std::vector<cv::Rect> opaque_cvrects;
+        std::fill(s.m_ops[i].opaque_cvrects.begin(), s.m_ops[i].opaque_cvrects.end(), cv::Rect(0, 0, 0, 0));
+        s.m_ops[i].opaque_cvrects.clear();
+    }
+    s.m_ops.clear();
+}
+
+void cleanupGSerializedDatas(GSerialized &s)
+{
+    //std::cout << "cleanupGSerializedDatas" <<  std::endl;
+    for (uint i = 0; i < s.m_datas.size(); i++)
+    {
+        //RcDesc rc;
+        s.m_datas[i].rc.id = 0;
+        s.m_datas[i].rc.shape = cv::GShape::GMAT;
+        //GMetaArg meta;
+        s.m_datas[i].meta = util::monostate();
+    }
+    s.m_datas.clear();
+}
```

Comment:
I still don't get why do you need all this clean-ups.

Reply:
This feature is currently not used. This was added for the test purposes when I tried in-place serialization. Desirialize in the same graph object.

Reply:
This is not needed